applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -2026,6 +2032,21 @@ GeometryData::IntegrationMethod SmallStrainUPwDiffOrderElement::GetIntegrationMe
     return GI_GAUSS;
 }
 
+std::vector<Vector> SmallStrainUPwDiffOrderElement::CalculateStrains(const std::vector<Matrix>& rDeformationGradients,
+                                                                     const std::vector<Matrix>& rBs,
+                                                                     const Vector& rDisplacements,
+                                                                     bool UseHenckyStrain) const
+{
+    std::vector<Vector> result;
+    std::transform(
+        rDeformationGradients.begin(), rDeformationGradients.end(), rBs.begin(), std::back_inserter(result),
+        [this, &rDisplacements, UseHenckyStrain](const auto& rDeformationGradient, const auto& rB) {
+        return CalculateStrain(rDeformationGradient, rB, rDisplacements, UseHenckyStrain);
+    });
+
+    return result;
+}
+
```

Comment:
I think it can be moved to utilities together with CalculateStrain and CalculateCauchyStrain. These functions look identical for UPw and DiffOrder elements. It can be next PR.